pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java

```diff
@@ -356,15 +354,7 @@ public long getProperty(String name, long defaultValue) {
    * @return the property String value. Fallback to default value if missing.
    */
   public String getProperty(String name, String defaultValue) {
-    Object rawProperty = getRawProperty(name, defaultValue);
-    if (rawProperty instanceof List) {
-      return StringUtils.join(((ArrayList) rawProperty).toArray(), ',');
-    } else {
-      if (rawProperty == null) {
-        return null;
-      }
-      return rawProperty.toString();
-    }
+    return _configuration.getString(relaxPropertyName(name), defaultValue);
```

Comment:
we should probably keep the original logic for this one? (maybe check null first) 
i don't think this change is needed as long as the previous bug fix is in. please double check

Reply:
If I understood that mapValue() method correctly, using `> 1` was on purpose. Basically, if a config has multi string values, those values are combined to a comma separated string; otherwise, return the raw property.

After discussion with @dd-willgan, I'm testing some ideas here: https://github.com/apache/pinot/pull/10785, mainly to keep things backward compatible as much as possible. Let's see how that one goes.